### DISCUSSION TITLE
Size limit of a vector in Move

### DISCUSSION THREAD

**Discussion by Alexzander-Stone** [aptos-labs/aptos-developer-discussions#87] [Questions] [answered]
Asked by `Robojosef` on Discord (#dev-discussions):
> What's the limit on the size of a vector?

**Keywords**
vector, size, gas

**Comment by Alexzander-Stone** [comment, marked as answer]
There isn't an explicit size limit for vectors. 

However, there **does** exist a practical limit caused by the [limited gas available to each TX](https://aptos.dev/concepts/gas-txn-fee#gas-parameters-set-by-governance).
In that, if the [vector that you're manipulating](https://aptos.dev/concepts/base-gas#vectors-1) is too large, you will run out of gas for your TX.

#### Useful information
* [Gas usage depends on computational cost of executing TXs](https://aptos.dev/reference/glossary/#gas:~:text=The%20gas%20required%20for%20a%20transaction%20depends%20on%20the%20size%20of%20the%20transaction%2C%20the%20computational%20cost%20of%20executing%20the%20transaction%2C%20and%20the%20amount%20of%20additional%20global%20state%20created%20by%20the%20transaction%20(e.g.%2C%20if%20new%20accounts%20are%20created).)
* [Basics of gas and storage fees](https://aptos.dev/concepts/gas-txn-fee)

**Reply by gregnazario** [reply]
Generally > 10,000 you'll want to use a SmartVector or SmartTable instead.  Also, keeping in mind iterating over 10k items can be computationally intensive
